### DISCUSSION TITLE
ElementDG() for ElementQuadP()

### DISCUSSION THREAD

**Discussion by Abelsylowlie** [kinnala/scikit-fem#990] [Q&A]
Hi, all. I try to use example 07 on quadrilateral mesh. It is OK for `ElementQuad1()` and `ElementQuad2()`.  But for `ElementQuadP()`, it gets this
![image](https://user-images.githubusercontent.com/95954250/206465590-828e1aea-5581-4a49-b8f3-12304b14da25.png)

I would like to know what causes this error and how to fix it. Thanks in advance.

**Comment by kinnala** [comment]
Thanks for the report. I think there is a bug in the DOF indexing. I should be able to fix it in few days.

**Reply by Abelsylowlie** [reply]
Thank you! I want to know the difference between `ElementQuadP()` and spectral element with nodal basis, since both of them use Legendre polynomial.

**Reply by kinnala** [reply]
Can you report which order you used for ElementQuadP?

**Reply by kinnala** [reply]
For me this works just fine:
```python
"""Discontinuous Galerkin method."""

from skfem import *
from skfem.helpers import grad, dot, jump
from skfem.models.poisson import laplace, unit_load

m = MeshQuad().refined(2)
e = ElementDG(ElementQuadP(7))
alpha = 1e-3

ib = Basis(m, e)
bb = FacetBasis(m, e)
fb = [InteriorFacetBasis(m, e, side=i) for i in [0, 1]]

@BilinearForm
def dgform(u, v, p):
    ju, jv = jump(p, u, v)
    h = p.h
    n = p.n
    return ju * jv / (alpha * h) - dot(grad(u), n) * jv - dot(grad(v), n) * ju

@BilinearForm
def nitscheform(u, v, p):
    h = p.h
    n = p.n
    return u * v / (alpha * h) - dot(grad(u), n) * v - dot(grad(v), n) * u

A = asm(laplace, ib)
B = asm(dgform, fb, fb)
C = asm(nitscheform, bb)
b = asm(unit_load, ib)

x = solve(A + B + C, b)

M, X = ib.refinterp(x, 4)

def visualize():
    from skfem.visuals.matplotlib import plot, draw
    ax = draw(M, boundaries_only=True)
    return plot(M, X, shading="gouraud", ax=ax, colorbar=True)

if __name__ == "__main__":
    visualize().show()
```
![Figure_1](https://user-images.githubusercontent.com/973268/206554021-649bb2b8-bd5c-436a-be72-90ba62cd90c9.png)

**Reply by Abelsylowlie** [reply]
It did not work using the above snippet for order >= 3 on my laptop with Windows 10 and scikit-fem-7.0.1.

**Reply by Abelsylowlie** [reply]
It works on the current unreleased version. Thanks.
